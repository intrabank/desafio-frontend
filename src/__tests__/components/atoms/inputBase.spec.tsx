import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputBase from '../../../components/atoms/inputBase';
import renderer from 'react-test-renderer';

describe('InputBase', () => {
	it('should render card component', () => {
		const tree = renderer
			.create(<InputBase state='idle' type='text' />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render error state', () => {
		render(<InputBase placeholder='test' state='error' type='text' />);
		const input = screen.getByPlaceholderText('test');

		expect(input).toBeInTheDocument();
	});

	it('should render active state', () => {
		render(<InputBase placeholder='test' state='active' type='text' />);
		const input = screen.getByPlaceholderText('test');

		expect(input).toBeInTheDocument();
	});

	it('should render whhen different  type', () => {
		render(<InputBase placeholder='test' state='active' type='password' />);
		const input = screen.getByPlaceholderText('test');

		expect(input).toBeInTheDocument();
	});
});
