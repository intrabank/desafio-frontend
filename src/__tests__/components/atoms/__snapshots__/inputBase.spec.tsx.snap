// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase should render card component 1`] = `
<input
  className="c-gpMnof c-gpMnof-gFpoIP-state-idle"
  type="text"
/>
`;
